backend/compact-connect/lambdas/python/purchases/handlers/privileges.py

```diff
@@ -211,21 +212,29 @@ def post_purchase_privileges(event: dict, context: LambdaContext):  # noqa: ARG0
     provider_id = _get_caller_provider_id_custom_attribute(event)
     user_provider_data = config.data_client.get_provider(compact=compact_abbr, provider_id=provider_id)
 
+    home_state_selection = ProviderRecordUtility.get_provider_home_state_selection(user_provider_data['items'])
+    if home_state_selection is None:
+        raise CCInternalException('No home state selection found for this user')
+
     license_records = ProviderRecordUtility.get_records_of_type(
         user_provider_data['items'],
         ProviderRecordType.LICENSE,
     )
 
-    home_state_selection = ProviderRecordUtility.get_provider_home_state_selection(user_provider_data['items'])
-    if home_state_selection is None:
-        raise CCInternalException('No home state selection found for this user')
-
-    home_state_license_record = ProviderRecordUtility.find_best_license(
-        license_records=license_records,
-        home_jurisdiction=home_state_selection,
+    # we now validate that the license type matches one of the license types from the home state license records
+    matching_license_record = next(
```

Comment:
You can combine this license record filtering step with your `ProviderRecordUtility.get_records_of_type()` call by leveraging its `_filter` kwarg.